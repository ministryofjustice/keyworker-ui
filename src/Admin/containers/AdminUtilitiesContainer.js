import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { setLoaded } from '../../redux/actions'
import Page from '../../Components/Page'
import { userType, configType } from '../../types'
import MessageBar from '../../MessageBar'

export class AdminUtilitiesContainer extends Component {
  componentWillMount() {
    const { setLoadedDispatch } = this.props

    setLoadedDispatch(true)
  }

  render() {
    const { user, config } = this.props
    const hasMaintainRolesAccess = user && (user.maintainAccess || user.maintainAccessAdmin)
    const showMaintainRoles = config && config.maintainRolesEnabled === 'true' && hasMaintainRolesAccess
    const showKeyworkerSettings = hasMaintainRolesAccess && user.migration

    return (
      <Page title="Admin and Utilities">
        <MessageBar {...this.props} />
        <div className="pure-g">
          <div className="pure-u-md-8-12">
            {!hasMaintainRolesAccess && !showMaintainRoles && !showKeyworkerSettings && (
              <p>There are no Admin or Utilities associated with your account.</p>
            )}

            {hasMaintainRolesAccess && (
              <div className="pure-u-md-6-12">
                <Link
                  id="enable_new_nomis_link"
                  title="Enable Nomis"
                  className="link"
                  to="/admin-utilities/give-nomis-access"
                >
                  Give access to New NOMIS
                </Link>
                <div className="padding-right-large">Allow prisons to use New NOMIS. Add new prison staff.</div>
              </div>
            )}
            {showKeyworkerSettings && (
              <div className="pure-u-md-5-12">
                <Link
                  id="keyworker_settings_link"
                  title="Key worker settings"
                  className="link"
                  to="/admin-utilities/manage-key-worker-settings"
                >
                  Manage key worker settings
                </Link>
                <div className="padding-right-large">
                  Allow auto-allocation. Edit key worker capacity and session frequency.
                </div>
              </div>
            )}
            {showMaintainRoles && (
              <div className="pure-u-md-5-12">
                <Link
                  id="maintain_roles_link"
                  title="Manage access roles"
                  className="link"
                  to="/admin-utilities/maintain-roles"
                >
                  Manage access roles
                </Link>
                <div className="padding-right-large">Add and remove staff roles.</div>
              </div>
            )}
          </div>
        </div>
      </Page>
    )
  }
}

AdminUtilitiesContainer.propTypes = {
  user: userType.isRequired,
  config: configType.isRequired,
  setLoadedDispatch: PropTypes.func.isRequired,
}

const mapStateToProps = state => ({
  user: state.app.user,
  config: state.app.config,
})

const mapDispatchToProps = dispatch => ({
  setLoadedDispatch: status => dispatch(setLoaded(status)),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminUtilitiesContainer)
