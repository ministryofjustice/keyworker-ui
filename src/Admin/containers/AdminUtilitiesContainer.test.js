import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import { AdminUtilitiesContainer } from './AdminUtilitiesContainer'

const props = {
  setLoadedDispatch: jest.fn(),
  user: {
    maintainAccessAdmin: false,
    activeCaseLoadId: '',
    caseLoadOptions: [],
    expiredFlag: false,
    firstName: 'Test',
    lastName: 'User',
    lockedFlag: false,
    maintainAccess: false,
    migration: false,
    staffId: 1,
    username: 'TestUser',
    writeAccess: true,
  },
  config: {
    keyworkerDashboardStatsEnabled: false,
    keyworkerProfileStatsEnabled: 'false',
    mailTo: 'email@test.com',
    notmEndpointUrl: '//notm.url',
    prisonStaffHubUrl: '//psh.url',
  },
  message: '',
}

describe('<AdminUtilitiesContainer />', () => {
  it('render with a message if there are no admin rights', () => {
    const wrapper = shallow(<AdminUtilitiesContainer {...props} />)

    expect(wrapper.find('p').text()).toEqual('There are no Admin or Utilities associated with your account.')
  })

  describe('render links', () => {
    it('should render a Link to the Give access to New NOMIS Admin section if user has maintainAccessAdmin role', () => {
      props.user.maintainAccessAdmin = true
      props.user.maintainAccess = false
      const wrapper = shallow(<AdminUtilitiesContainer {...props} />)
      const giveAccessLink = wrapper.find('Link').find({ to: '/admin-utilities/give-nomis-access' })

      expect(giveAccessLink.prop('children')).toEqual('Give access to New NOMIS')
    })

    it('should render a Link to Maintain Roles Admin section if user has maintain access role', () => {
      props.user.maintainAccess = true
      const wrapper = shallow(<AdminUtilitiesContainer {...props} />)
      const manageRolesLink = wrapper.find('Link').find({ to: '/admin-utilities/maintain-roles' })

      expect(manageRolesLink.prop('children')).toEqual('Manage access roles')
    })

    it('should render a Link to Manage key worker settings if user has migration and maintain access roles', () => {
      props.user.maintainAccess = true
      props.user.migration = true
      const wrapper = shallow(<AdminUtilitiesContainer {...props} />)
      const keyworkerSettingsLink = wrapper.find('Link').find({ to: '/admin-utilities/manage-key-worker-settings' })

      expect(keyworkerSettingsLink.prop('children')).toEqual('Manage key worker settings')
    })

    it('should render a Link to Maintain auth users if user has maintain auth user roles', () => {
      props.user.maintainAuthUsers = true
      const wrapper = shallow(<AdminUtilitiesContainer {...props} />)
      const keyworkerSettingsLink = wrapper.find('Link').find({ to: '/admin-utilities/maintain-auth-users' })

      expect(keyworkerSettingsLink.prop('children')).toEqual('Manage auth users')
    })
  })

  it('should render the admin and utilities page', () => {
    props.user.maintainAccessAdmin = true
    props.user.migration = true
    props.user.maintainAuthUsers = true
    const store = { subscribe: jest.fn(), dispatch: jest.fn(), getState: jest.fn(), setState: jest.fn() }
    store.getState.mockReturnValue({ app: { error: '', loaded: true } })

    const rendered = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <AdminUtilitiesContainer {...props} />
        </MemoryRouter>
      </Provider>
    )

    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
