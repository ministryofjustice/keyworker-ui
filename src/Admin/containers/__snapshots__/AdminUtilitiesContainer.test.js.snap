// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminUtilitiesContainer /> should render the admin and utilities page 1`] = `
Array [
  <div
    className="sc-dnqmqq ekasAr"
  >
    <ul
      className="sc-iwsKbI bhpoyq"
    >
      <li
        className="sc-gZMcBi gYqZNj"
      >
        <a
          data-qa="breadcrumb-home-page-link"
          href=""
        >
          Home
        </a>
      </li>
      <li
        className="sc-gZMcBi gYqZNj"
      >
        <span>
          Admin and utilities
        </span>
      </li>
    </ul>
  </div>,
  <div
    className="sc-htoDjs bCKyuL"
  >
    
    <h1
      className="sc-bdVaJa eLwxMn"
      size="LARGE"
    >
      Admin and utilities
    </h1>
    <div
      className="page-content"
    >
      <div
        className="sc-gqjmRU cthAvw"
      >
        <div
          className="sc-VigVT dUvdCr"
        >
          <a
            className="link"
            href="/admin-utilities/give-nomis-access"
            id="enable_new_nomis_link"
            onClick={[Function]}
            title="Enable Nomis"
          >
            Give access to New NOMIS
          </a>
          <div>
            Allow prisons to use New NOMIS. Add new prison staff.
          </div>
        </div>
        <div
          className="sc-VigVT dUvdCr"
        >
          <a
            className="link"
            href="/admin-utilities/manage-key-worker-settings"
            id="keyworker_settings_link"
            onClick={[Function]}
            title="Key worker settings"
          >
            Manage key worker settings
          </a>
          <div>
            Allow auto-allocation. Edit key worker capacity and session frequency.
          </div>
        </div>
        <div
          className="sc-VigVT dUvdCr"
        >
          <a
            className="link"
            href="/admin-utilities/maintain-roles"
            id="maintain_roles_link"
            onClick={[Function]}
            title="Manage access roles"
          >
            Manage access roles
          </a>
          <div>
            Add and remove staff roles.
          </div>
        </div>
        <div
          className="sc-VigVT dUvdCr"
        >
          <a
            className="link"
            href="/admin-utilities/maintain-auth-users"
            id="maintain_auth_users_link"
            onClick={[Function]}
            title="Manage auth users"
          >
            Manage auth users
          </a>
          <div>
            Maintain users that do not exist in NOMIS, only in auth.
          </div>
        </div>
        <div
          className="sc-VigVT dUvdCr"
        >
          <a
            className="link"
            href="/admin-utilities/create-auth-user"
            id="create_auth_user_link"
            onClick={[Function]}
            title="Create auth user"
          >
            Create auth user
          </a>
          <div>
            Create user in auth (not in NOMIS).
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
